From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Tue, 3 Dec 2024 12:14:04 +0000
Subject: [PATCH] Revert
 https://github.com/microsoft/terraform-provider-azuredevops/pull/1075

The validation always fails when running under the bridge because the bridge fills in the default values for values which are marked as ConflictsWith. Therefore, we'll remove this validation for the time being to unblock users from using the variable group resource.

This can be removed once https://github.com/pulumi/pulumi-terraform-bridge/issues/2618 is resolved.

Original customer issue: https://github.com/pulumi/pulumi-azuredevops/issues/514

diff --git a/azuredevops/internal/service/taskagent/resource_variable_group.go b/azuredevops/internal/service/taskagent/resource_variable_group.go
index f1d1cc18..7788bb97 100644
--- a/azuredevops/internal/service/taskagent/resource_variable_group.go
+++ b/azuredevops/internal/service/taskagent/resource_variable_group.go
@@ -381,21 +381,6 @@ func expandVariableGroupParameters(clients *client.AggregatedClient, d *schema.R
 	description := converter.String(d.Get(vgDescription).(string))
 	variables := d.Get(vgVariable).(*schema.Set).List()
 
-	// needed to detect if the secret_value attribute is set in the config
-	// see https://github.com/hashicorp/terraform-plugin-sdk/issues/741
-	for it := d.GetRawConfig().AsValueMap()[vgVariable].ElementIterator(); it.Next(); {
-		_, ctyVariable := it.Element()
-		ctyVariableAsMap := ctyVariable.AsValueMap()
-		name := ctyVariableAsMap[vgName].AsString()
-		valueSet := !ctyVariableAsMap[vgValue].IsNull()
-		secretValueSet := !ctyVariableAsMap[secretVgValue].IsNull()
-		isSecretSet := !ctyVariableAsMap[vgIsSecret].IsNull()
-
-		if valueSet && (secretValueSet || isSecretSet) || secretValueSet != isSecretSet {
-			return nil, nil, fmt.Errorf("`%s` variable can have either only `value` attribute or both `is_secret` and `secret_value` attributes", name)
-		}
-	}
-
 	variableMap := make(map[string]interface{})
 
 	for _, variable := range variables {
